scripts/remove_after_use/register_preprint_dois_with_crossref.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+import io
+import csv
+import sys
+import gzip
+import time
+import logging
+
+from website.app import setup_django
+setup_django()
+from django.core.paginator import Paginator
+
+from osf.models import PreprintService
+from osf_tests.factories import ProjectFactory, PreprintFactory, AuthUserFactory, PreprintProviderFactory
+
+from website import mails
+from website import settings
+from website.app import init_app
+from website.identifiers.clients import CrossRefClient
+
+
+logger = logging.getLogger(__name__)
+
+PAGE_SIZE = 5
+
+
+def register_existing_preprints_with_crossref(dry=True):
+    """This should do three things:
+    one, create a CSV of all preprints to be migrated for use by CNRI,
+    two, send metadata about the new preprint DOIs to CrossRef for registration,
+    three, add the new crossref specific DOIs to the preprint with the category 'doi' while marking the
+    legacy_doi identifier as deleted.
+    """
+    paginator = Paginator(
+        PreprintService.objects.filter(identifiers__category='legacy_doi', identifiers__deleted__isnull=True).order_by('pk'),
```

Comment:
The provider name is used in the metadata sent to crossref, so it'd be good to prefetch the provider with `select_related`.